### ISSUE TITLE
Chainhook predicate finds 93 occurrences of contract deploys for address but there are over 200 in explorer

### ISSUE THREAD

**Issue by timstackblock** [hirosystems/chainhook#357]
**Describe the bug**
When running the contract deploy predicate for address ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGMI only get 93 occurences returned but I can see there are over 200 contract deploys in explorer, filter for contract deploys in the activity list below to view the data.  https://explorer.hiro.so/address/ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM?chain=testnet

<img width="963" alt="Screen Shot 2023-07-28 at 12 34 16 PM" src="https://github.com/hirosystems/chainhook/assets/49165468/75b9ad52-1601-4e12-a3b6-bcc3b96138d9">

I was expecting all that data to be pulled in but chainhook scanned all 114982 blocks and only found 93 occurrences. 
<img width="780" alt="Screen Shot 2023-07-28 at 12 28 03 PM" src="https://github.com/hirosystems/chainhook/assets/49165468/d616c86e-65d3-4d76-91b0-02cb729c14b4">

There seems to be a large amount of contract deploys missing. Here is the predicate I selected the block height 110989 of the last contract deploy so all the data would be pulled in,  please let me know if there is something present in the file that needs changing and if this is a valid bug.

<img width="952" alt="Screen Shot 2023-07-28 at 12 35 43 PM" src="https://github.com/hirosystems/chainhook/assets/49165468/a081fd9c-208e-4535-942d-691de2343a33">

**Comment by lgalabru** [comment]
Thanks for the time spent on this @timstackblock, I think it's time for us to take on https://github.com/hirosystems/chainhook/issues/287.

**Comment by timstackblock** [comment]
Ok I have pushed up my test here https://github.com/hirosystems/chainhook/pull/347#issuecomment-1660404718  some are blocked by the corresponding bugs I found but I will comment on #287

**Comment by MicaiahReid** [comment]
I've done some digging, and I think this actually is okay! 

When a predicate is checked against a block, _all_ transactions in the block are checked to see if their deployer is the one provided in the predicate. If _any_ of the transactions match, we return the _whole_ block and that is _1_ occurrence. So, if a single block has multiple transaction deployments from the deployer, it will only count as one occurrence. However, if you inspect the actual block that is added to the file, it will have multiple transactions. Here is an example:

```JSON
{
  "chain": "stacks",
  "uuid": "a0eb4a64-c96c-4ce7-836f-1b0e004fb53f",
  "name": "Stacks Testnet Contract Deployment",
  "version": 1,
  "networks": {
    "testnet": {
      "start_block": 78395,
      "end_block": 78395,
      "if_this": {
        "scope": "contract_deployment",
        "deployer": "ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM"
      },
      "then_that": {
        "file_append": {
          "path": "./results/stacks-testnet-contract-deployment.json"
        }
      }
    }
  }
}
```

This block has 8 different contract deployments from the specified deployer, but it only results in one occurrence. So, though the block explorer shows many more deployments, there probably are the same amount.

To check this programmatically, maybe your tests can read in the resultant file, iterate over all blocks, and count all transactions with `metadata.kind.type == "ContractDeployment && metadata.kind.data.contract_identifier.startsWith(deployer)`?


---
Another interesting thing I found is that in the [explorer link you shared](https://explorer.hiro.so/address/ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM?chain=testnet), when you filter by contract deployer, there are some deployments by a _different_ address. I don't know why. But if you look at the `By ST1PQ...PGZGM` text under each deployment, you'll see that not all of them are actually from that address:
![image](https://github.com/hirosystems/chainhook/assets/17027138/fd8af078-9090-4fd5-94db-2a39ef3f0ad9)

So I only actually counted 177 occurrences of a contract deployment that were _actually_ from that address.

@timstackblock @lgalabru
